Source/WebGPU/WebGPU/Queue.mm

```diff
@@ -249,8 +291,12 @@ static bool validateWriteBufferInitial(size_t size)
     // FIXME(PERFORMANCE): Suballocate, so the common case doesn't need to hit the kernel.
     // FIXME(PERFORMANCE): Should this temporary buffer really be shared?
     id<MTLBuffer> temporaryBuffer = [m_device.device() newBufferWithBytes:data length:static_cast<NSUInteger>(size) options:MTLResourceStorageModeShared];
-    if (!temporaryBuffer)
+    if (!temporaryBuffer) {
+        ASSERT_NOT_REACHED();
         return;
+    }
+
+    didModifyResource(buffer.buffer());
```

Comment:
I don't think this is right. We don't want to synchronize literally every buffer that gets written to. It will be way more common for buffers to be written to from the CPU, but never read from the CPU, and this patch would cause lots of needless synchronization in that case. I think mapAsync() should schedule a synchronize call if one is necessary, rather than doing it unconditionally here.

Reply:
I think we still need the synchronization independent of mapAsync as the CTS test performs overlapping writes which can use memcpy or the blit encoder depending if the queue is idle or not.

Reply:
I am looking at the documentation more closely this should only be necessary for MTLStorageModeManaged so I need to determine where the race condition is on Apple Silicon. The test causing the race is this one:
```
    for (const { bufferOffset, data, arrayType, useArrayBuffer, dataOffset, dataSize } of writes) {
      const TypedArrayConstructor = globalThis[arrayType];
      const writeData = new TypedArrayConstructor(data);
      const writeSrc = useArrayBuffer ? writeData.buffer : writeData;
      this.queue.writeBuffer(buffer, bufferOffset, writeSrc, dataOffset, dataSize);
      memcpy(
      { src: writeSrc, start: dataOffset, length: dataSize },
      { dst: expectedData, start: bufferOffset });

    }

    this.debug(`expectedData: [${expectedData.join(', ')}]`);
    this.expectGPUBufferValuesEqual(buffer, expectedData);
```